_posts/2016-04-01-new-post.md

```diff
@@ -0,0 +1,912 @@
+---
+layout: post
+title: "Comparing Ember and React Components"
+description: ""
+category:
+tags: []
+---
+
+{% include JB/setup %}
+
+# Comment Box
+
+Fridays here at Greenfield are personal improvement days. We spend the day
+working on something of our own choice that will help us grow as developers. We
+spent part of a recent Friday going through the [React Tutorial][react-tutorial]
+as a team. Since we're all [Ember][ember] developers, many comparisons were
+made.
+
+The first time I went through the React Tutorial was over a year ago and, at the
+time, I had lots of feature envy. Fortunately, many of the awesome features that
+React has are now available in Ember-land.
+
+We're going to use Ember to build the same application so we can compare and
+contrast the way that both frameworks use components.
+
+## Getting Started
+
+We're going to use ember-cli to set up our app. This won't provide us with a
+local API server to run like in the source files provided by the React tutorial
+but we'll use another tool to handle that later on.
+
+If you don't already have ember-cli installed, head over to [Installing
+Ember][installing-ember].
+
+To get started, we'll use ember-cli to generate a new Ember app:
+
+```javascript
+$ ember new ember-comment-box
+```
+
+We can verify that our app is set up correctly by running the local ember server
+and viewing the app in our browser [http://localhost:4200](localhost):
+
+```bash
+$ cd ember-comment-box
+$ ember server
+$ open http://localhost:4200
+```
+
+## Recap
+
+The main features of the app are:
+
+- A list of comments.
+- A form for adding new comments.
+- Load comment data from a server.
+- Implement optimistic updates.
+
+Let's take a look at our component hierarchy:
+
+```no-highlight
+- comment-box
+  - comment-list
+    - comment-list-item
+  - comment-form
+```
+
+We're going to have a `comment-box` component that has two child components,
+`comment-list` and `comment-form`. The `comment-list` component will also have
+many child `comment-list-item` components.
+
+--- do a diagram! ---
```

Comment:
Oh you mean like with borders showing what is in what? Yea that would be useful. Then you could replace

+- comment-box
 +  - comment-list
 +    - comment-list-item
 +  - comment-form

With the diagram